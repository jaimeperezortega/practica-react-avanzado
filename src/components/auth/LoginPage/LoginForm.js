import T from 'prop-types';

import useForm from '../../../hooks/useForm';

const validEmail = ({ email }) => email;
const validPassword = ({ password }) => password;

function LoginForm({ onSubmit }) {
  const {
    formValue: credentials,
    handleChange,
    handleSubmit,
    validate,
  } = useForm({
    email: '',
    password: '',
    remember: false,
  });
  const { email, password, remember } = credentials;

  return (
    <form className="loginForm" onSubmit={handleSubmit(onSubmit)}>
      <input className='loginForm-field' name="email" value={email} onChange={handleChange} />
      <input
        type="password"
        className= 'loginForm-field'
        name="password"
        value={password}
        onChange={handleChange}
      />
      <input
        type="checkbox"
        name="remember"
        checked={remember}
        onChange={handleChange}
      />
      <button disabled={!validate(validEmail, validPassword)}>Login</button>
    </form>
  );
}

LoginForm.propTypes = {
  onSubmit: T.func.isRequired,
};

export default LoginForm;
