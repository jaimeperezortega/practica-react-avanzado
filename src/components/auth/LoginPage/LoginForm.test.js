import React from 'react';
import { shallow } from 'enzyme';

import LoginForm from './LoginForm';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('LoginForm', () => {
  const props = {
    isLoading: false,
    onSubmit: jest.fn(),
  };

  const render = () => shallow(<LoginForm {...props} />);

  test('should render', () => {
    const wrapper = render();
    expect(wrapper.exists()).toBe(true);
  });

  test('snapshot testing', () =>{
    const wrapper = render();
    expect(wrapper).toMatchSnapshot();
});

  test('snapshot testing withisLoading', () =>{
      const wrapper = render();
      wrapper.setProps({isLoading:true})
      expect(wrapper).toMatchSnapshot();
  });

  test('should submit credentials', () => {
    const wrapper = render();
   

    const usernameField = wrapper.find('.loginForm-field').at(0);
    usernameField.props().onChange({target: {name: 'username', value: 'jaime'}})

    const passwordField = wrapper.find('.loginForm-field').at(1);
    passwordField.props().onChange({target: {name: 'password', value: 'password'}})

    const form = wrapper.find('.loginForm');
    form.simulate('submit',{preventDefault:() => {}});

    expect(props.onSubmit).toHaveBeenCalled()
  });
})