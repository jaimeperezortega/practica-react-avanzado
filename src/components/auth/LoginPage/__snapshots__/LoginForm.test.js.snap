// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm snapshot testing 1`] = `
<form
  className="loginForm"
  onSubmit={[Function]}
>
  <input
    className="loginForm-field"
    name="email"
    onChange={[Function]}
    value=""
  />
  <input
    className="loginForm-field"
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <input
    checked={false}
    name="remember"
    onChange={[Function]}
    type="checkbox"
  />
  <button
    disabled={true}
  >
    Login
  </button>
</form>
`;

exports[`LoginForm snapshot testing withisLoading 1`] = `
<form
  className="loginForm"
  onSubmit={[Function]}
>
  <input
    className="loginForm-field"
    name="email"
    onChange={[Function]}
    value=""
  />
  <input
    className="loginForm-field"
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <input
    checked={false}
    name="remember"
    onChange={[Function]}
    type="checkbox"
  />
  <button
    disabled={true}
  >
    Login
  </button>
</form>
`;
